### ISSUE TITLE
Clarity on use cases for configuring media types

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#737]
I'm having trouble understanding what the most important use cases are for configuring media types. I can identify the following:

- I want my Islandora repository to manage files that have a ".foo" extension, and adding this extension to the existing "File" media type is not going to work for me. So I create a new media type, "Foo media", and now I need to tell Workbench that any files it sees with that extension in the 'file' or an 'additional_file' column should be ingested into media of type "Foo media".
- I want my Islandora repository to manage files that have a ".foo" extension, but I don't have any special requirements that the existing "File" media type can't handle. So all I need to do is tell Workbench that any files it sees with that extension in the 'file' or an 'additional_file' column should be ingested into media of type "File" media.

What are some other use cases Workbench needs to accommodate? Are there any use cases that describe *overriding* (i.e. changing) a default extension->media type mapping, for example? @rosiel and @Natkeeran, I know you've been bitten by the `media_type_file_fields` and related config settings described at https://mjordan.github.io/islandora_workbench_docs/media_types/. I am interested in hearing everyone's use cases so we can make this aspect of Workbench better. I realize that docs page is terrible, but I'm not interested in feedback on that page at the moment; what I need to understand first before we improve things is the kinds of things we expect to be able to do with Workbench.

**Comment by rosiel** [comment]
Hi Mark,

I technically have a third case but I split it up into separate migrations to avoid the issue (and use the option in the first case).

I have multiple PDF files - some are private service files and we put them in Document. Other PDFs are "signature files" which give the library the right to display the item, and we put them in a separate, "Private Media" type that defaults to unpublished and stores files in the private filesystem. Using (published/unpublished) as a method of access control may not be the most robust thing, but it works. So in order to do this in workbench, i make a separate Workbench CSV of just the private files and configure it to use our custom media type. It works, and I don't feel the need to request the ability to do column-by-column mappings (that's just too hard for everyone; making a separate workbench ingest is easy enough).

I can easily imagine (though maybe this is better done in unison with the whole community?) adding two new media types, PDF and Large Image, which would remove the need for Viewer Overrides and would also change the default mappings for .pdf, .tiff, and .jp2. But if I implement this before the rest of the community, then I'd want Case 1 available.

I thought that by default, Workbench takes any unknown file extension and by default tries to put it in File (if allowed). I think that's a great default behaviour.

**Comment by mjordan** [comment]
Thanks @rosiel. 

> I thought that by default, Workbench takes any unknown file extension and by default tries to put it in File (if allowed). I think that's a great default behaviour.

Yes, this is the intended behavior. Do you see that not happening?

Am I correct in understanding that the first use case I outline above is sufficient and that by splitting up your job as you describe is OK? If so, maybe your "third use case" could be a recipe? I'm thinking of asking the community to contribute recipes or novel ways they've used Workbench, maybe as part of an Open Meeting.

**Comment by rosiel** [comment]
Yes, the first use case - being able to assign an already-mapped extension to a new media type - has been sufficient for my needs so far.

I think the second use case - a new file extension that should be mapped to an existing media type  - is also good and useful. For example, right now there are [lots of video types that can go in video](https://github.com/Islandora-Devops/islandora-starter-site/blob/main/config/sync/field.field.media.video.field_media_video_file.yml#L31), but so far workbench only knows .mp4. Having a way to map these (or any other video file extensions I find that also work with Drupal's Video media type) to Video is excellent.

**Comment by mjordan** [comment]
Thanks very much. And thanks for the heads up on the video file types, I'll update Workbench's defaults.
